de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java

```diff
@@ -1666,4 +1668,24 @@ public void invalidateUninvoicedFreightCostCandidate(@NonNull final OrderId orde
 
 		invalidateCandsFor(freightCostCandQueryBuilder);
 	}
+	@Override
+	public IQuery<I_C_Invoice_Candidate> createICQueryBuilder(List<ExternalHeaderAndLineId> headerAndLineIds) {
```

Comment:
introduce proper methods like
* createSelectionByHeaderAndLineIds(...)
* retrieveByHeaderAndLineIds(...)

rename `createICQueryBuilder` to `query` and make it private.

Reply:
done.